README.md
# sediROM for the Samsung Captivate Glide

(codenames: SGH-I927, i927)

All scripts needed to run sediROM on the Cappy.
Checkout the "TAGS" tab in github to find a specific sediROM release!

Help & Support: 

    http://forum.xda-developers.com/showthread.php?t=2789727

    If you are not yet allowed to write / answer within the dev area go here instead:
    http://forum.xda-developers.com/showthread.php?t=2789119


Flashing Guide:

    1. I'M NOT RESPONSIBLE IF YOU BRICK YOUR PHONE (nor the things that happens to your house and/or car!)
    2. Copy BOTH the sediROM zip AND the sediROM md5 file to your Glide!!
    3. Boot into Recovery mode (Poweroff the device then Power on while holding Volume Down)
    4. Ensure that you use TWRP(!) and that the MD5 sum file is in the same directory as the sediROM zip! Only then TWRP will automatically check the MD5 !
    5. Flash the latest sediROM zip file
    6. Choose "Easy Installation" mode
    7. Answer the few questions and wait until the flashing finishes
    
    OPTIONAL: You can decide to change kernel, baseband etc before rebooting from AROMA.
    Simply choose the related option after the easy installation has finished.
    
    8. When finished reboot and wait until it has fully started up
    
    OPTIONAL: You can choose to NOT reboot here and restart the wizard again. That way you can install more apps or change the Kernel etc.
    To do so simply select to install the sediROM zip again after you finished the easy installation and instead of step 6 choose the "Modify" mode.
    Afterwards go on with step 7 to 13.

    9. The Android setup wizard should come up (if not ­> flash once again !)
    10. Go through the wizard and reboot once again afterwards
    11. Enjoy! and hit my thanks button whereever you see a post of me ;)
    12. Please read the FAQ (http://forum.xda-developers.com/showpost.php?p=54075993&postcount=2)!
    13. Please file a bug if you encounter problems: File a bug (http://forum.xda-developers.com/devdb/project/?id=4942#bugReporter)


CHANGES

v2.0.64 -v2.0.1 (Release date: 2015-11-03)
--------------------------------------------------
Major Release

Github detailed changelog (compared with the previous version):
https://github.com/xdajog/android_i927_sediROM/compare/v1.7...v2.0

Github tag for this version:
https://github.com/xdajog/android_i927_sediROM/tree/v2.0

[U]Enhancements[/U]
[LIST]
[*]first sediROM app (sediROM_boot.apk) inside.. the app itself is extremely simple: a text and a button thats all. When
sediROM boots the first time a new added init script will detect if this is the first boot and if this is the case it will open
the sediROM_boot app. Read & follow carefully the hints there!
you to reboot. May be annoying but due to douzends of changes in v2.0 it is really necessary to point to a clean reboot.
[*]All scripts related to run sediROM on github now !!!! Starting from v1.7.
[*]Introduced the first version of sediKERNEL (v1.0) a customized kernel optimized for STOCK ICS so as for sediROM.
sediKERNEL is based on CM11 kernel made by bubor (so l2_hsic patched, no wakeup delays, OC etc) enhanced by:
- adding MTP support for STOCK ICS!
- less battery drain
[*]default kernel = sediKERNEL v1.0
[*]Upgraded AROMA from v2.56 to v2.70-RC2 (means compiling 2.70rc2 from the sources!)
[*]Go DIRECTLY from the installer to the MODIFY mode!
That means when you choosen the easy installation mode and everything went fine you will get the
offer to open the modify mode instead of rebooting
[*]No adb debugging from initial ram disk (security fix)
[*]No adb debugging as default (security fix)
[*]Integrated LiteKernel_l2hsicpatched-bubor-r20150506 with L2_HSIC patch included (hopefully fix battery drain)
The kernel is based on code of 2014-04 afaik also maded by bubor
[*]Integrated enhanced UCLJ3 stock Kernel (option in modify mode)
enhancements:
- init.d support (this enables sediROM bluetooth fix for this kernel)
- added custom boot animation support
[*]Migrated and integrated JB 4.2 LiteKernel release-20130221 to sediROM (option in modify mode) which comes with the following changelog:
(all changes between v1.2.2 to 20130221)
- Interactive set as default governor ... Wheatley lags on AOSP
- Added FM Radio Driver
- Fixed Mic for AOSP
- Fixed/Added 1.4ghz frequency
- Fixed/Added Smartassv2
- Removed USB Whitlists
- Recoded BLN myself so that there is no need for an app... has in kernel blinking ... 
- Tons of Cleanup
- XZ Kernel Compression
- Removed Wake Lag
- Fixed and increased Charging Current
- Tweaked Ondemand for better performance
- New Storage Setup
[*]Migrated and integrated JB 4.2 LiteKernel release-20130222 (option in modify mode) which comes with the following changelog.
HINT: MTP does not work with this kernel. I will not fix that! If you need MTP use release-20130221 or sediKERNEL!
(all changes between r20130221 to r20130222)
- Entropy Tweaks inspired by lambgx02s Seeder (for silky smoothness)
- Memory Managment Tweaks
- Added Dynamic vsync
- Zipaligning and Fix permissions at boot (zeppelinrox script)
- Tons of Kernel Tweaks for Battery life and Performance...
- IO tweaks...
- Auto EFS Backup...
- New Experimental WIFI Management battery saving feature: If at screen off, WIFI is inactive and or using very little traffic, it gets turned off
  and then on again once you wake device, if battery level is below 50% it will no longer turn wifi on again, if you disable WIFI it will be left alone...
- Decreased Vibration Intensity (when you boot up device you will feel a slight vibration)...
- New Experimental CPU Management feature: Frequency get changed according to battery level....
- No more laggy lock screen drawing ... 
- Instant wake to lock screen
- Removed Increase Charging Current mod till more testing can be done
- Improvements for better battery life
- Stability
- Option to Disable WIFI and CPU Control
- Clear Memory after boot
- Massive Improvements to: Performance, Battery Life
- Fixed Battery leak with CPU + WIFI manager
- Memory Management Improvements 
- SD card Speed Tweaks
- Reduced Wake Lag
- Reduced Stuttering when playing music on screen off
- CPU-Manager is now enabled by default ... it boosts wake speed manages sleep speed and reduces max speed as your battery diminishes ... the profiles are fully modifiable and all with no battery drain ... 
- MTP is now the default pc connection ... if you want mass storage change /data/LiteKernel/MTP to "0"
- Memory management improved .... should also help for battery life
- frequency with Interactive governor will now stay a little lower ... should help with screen on battery life ....
[*]crond (provided by busybox) activated to automatize things like in Linux
init script 00sediROM will prepare the settings, paths etc for crond to start
and init script 99sediROM will start the crond
[*]Open Camera will be the only camera app installed by default. HD Ultra camera stays an option in modify mode.
[*]added this changelog to AROMA installer screen
[*]added time and date to sediROM dmesg output
[*]added modify option after easy install in README of the installer zip
[*]updated sqlite3 binary to v3.8.7.4 (THANKS to user tech128 details: http://forum.xda-developers.com/showpost.php?p=52174054)
[*]Removed all my own copyright hints and licensed all sediROM scripts under CC BY-SA 4.0 (http://creativecommons.org/licenses/by-sa/4.0) license! Freedom for everyone ;o)
[*]New script header including new version concept of all sediROM scripts
[*]Updated SuperSU app and binaries to v2.46
[*]Installation of SuperSU is now completely based on the original installer to avoid any problems while installing
[*]Added a new minimalistic terminal app AndroidTerm (https://play.google.com/store/apps/details?id=jackpal.androidterm) which replaces connectbot as default installed terminal app.
[*]Connectbot will not being installed by default anymore but you can still install it in modify mode.

packaging/package_sediROM.sh
#!/bin/bash
#######################################
#
#
# Package sediROM helper
#
# v2016-01-04
#######################################
ANSW=undef
LVERFILE=package_last-version.txt
WORKDIR="../../PROJECT_sediROM"
BDIR="$PWD"

# check for vendor dev specification
VENDEV="$1"
while [ -z "$VENDEV" ];do
    echo
    echo "sorry you haven't specified a device name!"
    echo
    echo "Please type in now (e.g.: i927):"
    echo
    read VENDEV
done

F_TYPE(){
    echo
    echo "Type in NIGHTLY, BETA, RELEASE or TEST"
    echo "or press ENTER for TYPE = 'NIGHTLY'!"
    echo
    echo "  TEST    = META-INF only"
    echo "  NIGHTLY = All stuff + MD5"
    echo "  BETA    = same as NIGHTLY but with BETA in name"
    echo "  RELEASE = All stuff + MD5 + Create Release Package"
    echo
    echo "Your choice: "
    read TYPE
    [ -z "$TYPE" ]&&TYPE=NIGHTLY
}

# get last remembered version
LAST_VER(){
    if [ -r "$LVERFILE" ];then
        PREVVER=$(cat $LVERFILE)
        # split major and minor version number
        MINOR=$(echo ${PREVVER##*\.})
        MAJOR=$(echo ${PREVVER%\.*})
        # count it by 1
        NEWMINOR=$(($MINOR + 1))
        OFFERVER=$(echo $MAJOR.$NEWMINOR)
    else
        echo no previous version detected
    fi
}

# set current version
SET_LASTVER(){
    VERSION=$1
    echo $VERSION > $LVERFILE
}

TYPE_PROOF(){
    #debug
    #echo "DEBUG: in TYPE_PROOF"

    unset PROOF_STAT
    if [ -z "$TYPE" ];then
        PROOF_STAT=1
        F_TYPE
    else
        if [ $TYPE != "NIGHTLY" ];then
            if [ $TYPE != "TEST" ];then
                if [ $TYPE != "BETA" ];then
                    if [ $TYPE != "RELEASE" ];then
                        PROOF_STAT=1
                        F_TYPE
                    else
                        PROOF_STAT=0
                    fi
                else
                    PROOF_STAT=0
                fi
            else
                PROOF_STAT=0
            fi
        else
            PROOF_STAT=0
        fi   
    fi
    return $PROOF_STAT
}

while [ $ANSW != "y" ];do
    clear
    if [ -z "$ANSW" ];then break ;fi
    unset ANSW VERSION TYPE
    PROOF_STAT=1

    echo
    echo "$VENDEV mode ..."
    echo

    LAST_VER
    echo 
    echo "Previous version was $PREVVER."
    echo "Should we use >$OFFERVER< for this build?"
    echo
    echo "Type in a new version (e.g.: 'v1.5.0')"
    echo "or simply press ENTER for using $OFFERVER."
    echo
    read UVERSION
    if [ -z $UVERSION ];then
        echo using offered version
        VERSION=$OFFERVER
    else
        echo using your version
        VERSION=$UVERSION
    fi

    while [ -z $VERSION ];do
    	echo "Enter version number (e.g.: 'v1.5.0'):"
    	read VERSION
    done

    while [ "$PROOF_STAT" -ne "0" ];do
            TYPE_PROOF $TYPE
    done

    while [ -z "$ANSW" ];do
        clear
    	echo
    	echo "Your choice was:"
    	echo "	VERSION	= $VERSION"
    	echo "	TYPE	= $TYPE"
    	echo
    	echo "Is that OK?" 
        echo "Press -RETURN- or type 'y' for yes."
        echo "To ABORT instead type in 'abort' and we will starting over again:"
        read ANSW
        if [ -z $ANSW ];then ANSW="y" ;fi
    done 
done

# everything fine so do the magic

case "$TYPE" in
	TEST)
        # add vendor dev spec to type
        TYPE="${VENDEV}_${TYPE}"

        echo "$TYPE zip will be created."
		cd $WORKDIR && zip -x "*gitignore" -y -q -r -5 sediROM_${TYPE}_${VERSION}.zip META-INF && cd $BDIR
        if [ $? -eq 0 ];then
            echo "$TYPE: Completed successfully. This window will autoclose in 2 seconds."
            sleep 2 && echo CLOSING NOW && exit
        else
            echo "ERROR occured while creating ZIP file ${VERSION}.zip"
        fi
	;;
	BETA|NIGHTLY)
        # add vendor dev spec to type
        TYPE="${VENDEV}_${TYPE}"

        echo "$TYPE zip will be created. This will take a long time"
        cd $WORKDIR && zip -x "*gitignore" -y -q -r -9 sediROM_${TYPE}_${VERSION}.zip boot.img customize META-INF modem.bin setup system tmp README.txt \
            && echo "Creating MD5.." && md5sum sediROM_${TYPE}_${VERSION}.zip > sediROM_${TYPE}_${VERSION}.zip.md5 \
            && echo "MD5 created successfully." \
            && cd $BDIR
        if [ $? -eq 0 ];then
            SET_LASTVER "$VERSION"
            echo "$TYPE: Completed successfully. This window will close in 30 seconds."
            sleep 10 && echo ... 20 seconds
            sleep 10 && echo ... 10 seconds
            sleep 10 && echo CLOSING NOW && exit
        else
            echo "ERROR occured while creating ZIP file ${VERSION}.zip"
            read DUMMY
        fi
	;;
	RELEASE)
        # add vendor dev spec to type
        TYPE="${VENDEV}_${TYPE}"

        echo "$TYPE zip will be created. This will take a very VERY long time..."
        cd $WORKDIR && zip -x "*gitignore" -y -q -r -9 sediROM_${VERSION}.zip boot.img customize META-INF modem.bin setup system tmp README.txt && cd $BDIR
        if [ $? -eq 0 ];then 
            echo "Zip completed. Creating sediROM $TYPE Package..."
            cd $WORKDIR && zip -T sediROM_${VERSION}.zip \
                && echo "ZIP integrity OK. Creating MD5.." && md5sum sediROM_${VERSION}.zip > sediROM_${VERSION}.zip.md5 \
                && echo "MD5 created successfully. Packaging whole stuff now.." && cp tmp/CHANGES . && zip -r -0 sediROM_${MAJOR}_EXTRACT-ME-BEFORE-FLASHING.zip sediROM_${VERSION}.zip sediROM_${VERSION}.zip.md5 README.txt CHANGES\
                && echo "Release Packaging finished. Cleaning up..." && rm -f sediROM_${VERSION}.zip sediROM_${VERSION}.zip.md5 CHANGES \
                && cd $BDIR
            if [ $? -eq 0 ];then
                SET_LASTVER "$VERSION"
                echo "$TYPE: Completed successfully. This window will autoclose in 30 seconds."
                sleep 10 && echo ... 20 seconds
                sleep 10 && echo ... 10 seconds
                sleep 10 && echo CLOSING NOW && exit
            else
               echo "$TYPE: ERROR occured while creating ZIP file ${VERSION}.zip"
            fi 
        else 
            echo "ERROR occured while creating ZIP file ${VERSION}.zip"
        fi
	;;
	*)
        echo "FATAL ERROR. THAT SHOULD NEVER HAPPEN!"
        read DUMMY
    ;;
esac

SET_LASTVER "$VERSION"


system/xbin/sedi_btFCdetect.sh
#!/system/bin/sh
#############################################################################################################################
#
# Author:   Thomas Fischer (xdajog) <mail@se-di.de>
# Desc:		This code is part of sediROM (http://forum.xda-developers.com/showthread.php?t=2789727)
# Version:  v2.5 (the first digit reflects the sediROM major version when this file has been created/changed/modified)
#
# License:  This code is licensed under the Creative Commons License CC BY-SA 4.0.
#
# DISCLAIMER:
# The following deed highlights only some of the key features and terms of the actual license.
# It is NOT a license and has NO legal value. You should carefully review ALL of the terms and conditions of the actual
# license before using the licensed material.
#
# Please check the following link for details and the full legal content:
# http://creativecommons.org/licenses/by-sa/4.0/legalcode
#
#    You are free to:
#
#       Share — copy and redistribute the material in any medium or format
#       Adapt — remix, transform, and build upon the material
#
#       for any purpose, even commercially.
#       The licensor cannot revoke these freedoms as long as you follow the license terms.
#
#    Under the following terms:
#
#       Attribution:
#          You must give appropriate credit, provide a link to the license, and indicate if changes were made.
#          You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
#
#       ShareAlike:
#          If you remix, transform, or build upon the material, you must distribute your contributions under the same
#          license as the original.
#
#############################################################################################################################

echo "$(date +%D_%T) - sediROM($0): starting. Should be executed by cron." >> /dev/kmsg

# path where the config and log will be saved (may be auto adapted later in this script)
SEDIDIR=/preload/.sediROM/

BTIPATH=$(dirname /data/misc/bluetoothd/*/.)
BTINDI="$BTIPATH/btfix.indicator"

# Indicator file will be added by shutdown script and that requires at least 1 clean shutdown.
IFLAG=$SEDIDIR/.initialized

############################################################
# ensure that /data is not encrypted and if so we wait until it will
# be unlocked by the user
ENCSTATE=$(getprop vold.post_fs_data_done)

# ensure that system has booted completely
BOOTSTATE=$(getprop sys.boot_completed)

while [ ! -f "$IFLAG" ] || [ ! "x$ENCSTATE" == "x1" ] || [ ! "x$BOOTSTATE" == "x1" ];do
    echo "$(date +%D_%T) - sediROM($0): /data (still) locked or sediROM is not fully initialized." >> /dev/kmsg
    RETIFLAG=$(test -f "$IFLAG")
    echo "$(date +%D_%T) - sediROM($0): boot state was: $BOOTSTATE (should be empty or 1)." >> /dev/kmsg
    echo "$(date +%D_%T) - sediROM($0): $IFLAG test was: $RETIFLAG (should be empty or 0)." >> /dev/kmsg
    echo "$(date +%D_%T) - sediROM($0): Encryption test was: $ENCSTATE (should be 1)." >> /dev/kmsg
    sleep 10s
    # re-check states
    ENCSTATE=$(getprop vold.post_fs_data_done)
    BOOTSTATE=$(getprop sys.boot_completed)
    
    # ensure sdcard was initialized
    while [ ! -d "/sdcard/.sediROM" ];do sleep 3s ;done
    
    # if we were not able to use preload as log storage we will use sdcard instead    
    SEDISDDIR=/sdcard/.sediROM # if we cannot mount /preload this is a fallback
    INDIMOVED=$SEDISDDIR/dir-moved-2-preload.txt # indicator file to know if we are using preload or not               
    if [ ! -f "$INDIMOVED" ];then                                                                              
        SEDIDIR=$SEDISDDIR
    fi                                                                                                         
    echo "$(date +%D_%T) - sediROM($0): initflag dir set to $SEDIDIR" >> /dev/kmsg
    IFLAG=$SEDIDIR/.initialized
done

# set working dir according to the above result
WRKDIR=$SEDIDIR/bin
LOG=$WRKDIR/${0##*/}.log

# setup work dir
[ ! -d $WRKDIR ] && mkdir -p $WRKDIR

# check if a force close / soft reboot has taken place
# and if so re-enable bluetooth
if [ ! -f "$BTINDI" ];then
    echo "$(date +%D_%T) - sediROM($0): sediROM fully initialized. Exec 92sediROM_btfix again after FC." >> /dev/kmsg
    echo "$(date) sediROM: starting $0" > $LOG
    sh /etc/init.d/92sediROM_btfix >> $LOG
else
    echo "$(date) sediROM: $0 - BT indicator file is here. nothing to do." > $LOG
fi
echo "$(date +%D_%T) - sediROM($0): finished." >> /dev/kmsg


META-INF/com/google/android/aroma/efsbakpath.sh
#!/sbin/sh
#############################################################################################################################
#
# Author:   Thomas Fischer (xdajog) <mail@se-di.de>
# Desc:		This code is part of sediROM (http://forum.xda-developers.com/showthread.php?t=2789727)
# Version:  v2.0 (the first digit reflects the sediROM major version when this file has been created/changed/modified)
#
# License:  This code is licensed under the Creative Commons License CC BY-SA 4.0.
#
# DISCLAIMER:
# The following deed highlights only some of the key features and terms of the actual license.
# It is NOT a license and has NO legal value. You should carefully review ALL of the terms and conditions of the actual
# license before using the licensed material.
#
# Please check the following link for details and the full legal content:
# http://creativecommons.org/licenses/by-sa/4.0/legalcode
#
#    You are free to:
#
#       Share — copy and redistribute the material in any medium or format
#       Adapt — remix, transform, and build upon the material
#
#       for any purpose, even commercially.
#       The licensor cannot revoke these freedoms as long as you follow the license terms.
#
#    Under the following terms:
#
#       Attribution:
#          You must give appropriate credit, provide a link to the license, and indicate if changes were made.
#          You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
#
#       ShareAlike:
#          If you remix, transform, or build upon the material, you must distribute your contributions under the same
#          license as the original.
#
#############################################################################################################################
echo "/sdcard/efs_$(date +%F_%s 2>&1 | /sbin/busybox grep -v bionic).dd"

system/xbin/sediROM_testsuite.sh
#!/system/bin/sh
#############################################################################################################################
#
# Author:   Thomas Fischer (xdajog) <mail@se-di.de>
# Desc:		This code is part of sediROM (http://forum.xda-developers.com/showthread.php?t=2789727)
# Version:  v2.0 (the first digit reflects the sediROM major version when this file has been created/changed/modified)
#
# License:  This code is licensed under the Creative Commons License CC BY-SA 4.0.
#
# DISCLAIMER:
# The following deed highlights only some of the key features and terms of the actual license.
# It is NOT a license and has NO legal value. You should carefully review ALL of the terms and conditions of the actual
# license before using the licensed material.
#
# Please check the following link for details and the full legal content:
# http://creativecommons.org/licenses/by-sa/4.0/legalcode
#
#    You are free to:
#
#       Share — copy and redistribute the material in any medium or format
#       Adapt — remix, transform, and build upon the material
#
#       for any purpose, even commercially.
#       The licensor cannot revoke these freedoms as long as you follow the license terms.
#
#    Under the following terms:
#
#       Attribution:
#          You must give appropriate credit, provide a link to the license, and indicate if changes were made.
#          You may do so in any reasonable manner, but not in any way that suggests the licensor endorses you or your use.
#
#       ShareAlike:
#          If you remix, transform, or build upon the material, you must distribute your contributions under the same
#          license as the original.
#
#############################################################################################################################

# the first sediROM testing suite!
# execute it like this:
# adb push sediROM_testsuite.sh /sdcard/ && adb shell "su -c sh /sdcard/sediROM_testsuite.sh"

LOG=/sdcard/testsuite.log

echo "Starting testsuite: $(date +"%F %T")" > $LOG

# hard aborting/failures
F_FAIL(){
    echo -e "-> !!!!! ERROR !!!!! : While processing test $1. ABORTED\n"
    echo "ERROR: While processing test $1. ABORTED." >> $LOG
    exit 2
}

# soft failures without aborting
F_WARN(){
    echo -e "!!!!! WARNING !!!!!: While processing test $1.\nResult was: >$2<"
    echo "WARNING: While processing test $1.\nResult was: >$2<" >> $LOG
    echo 
}

# echo output with variable as result
END_VAR(){
    echo "-> OK: Testing $1 finished successfully"
    echo "-> OK: Testing $1 finished successfully" >> $LOG
    echo -e "-> Result was:\n$2\n"
}

# echo output with temp file as result
END_FILE(){
    echo "-> OK: Testing $1 finished successfully"
    echo "-> OK: Testing $1 finished successfully" >> $LOG
    echo -e "-> Result was:\n$(cat /sdcard/test)\n\n"
}
###########################################################################

echo -e "\n\nStarting the sediROM test suite! Have fun ;-)"
echo -e "-------------------------------------------------------\n"

echo "\n****************************************************"
TEST=">crond<"
echo "-> Testing $TEST"
ps |grep crond > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">crond config<"
echo "-> Testing $TEST"
FSIZE=$(stat -t /system/etc/cron.d/root |cut -d " " -f 2 )
if [ "$FSIZE" -lt 10 ];then
    stat -t /system/etc/cron.d/root
    F_FAIL "$TEST"
else
    END_VAR "$TEST" "$FSIZE"
fi

echo "\n****************************************************"
TEST=">sediROM app<"
echo "-> Testing $TEST"
SAPP=/system/app/sediROM_boot.apk
stat $SAPP > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">easy install: sediKERNEL<"
echo "-> Testing $TEST"
KERNEL=$(uname -r |grep -o sediKERNEL)
if [ "$KERNEL" != "sediKERNEL" ];then
    F_WARN "$TEST" "$KERNEL"
else
    END_VAR "$TEST" "$KERNEL"
fi

echo "\n****************************************************"
TEST=">easy install: gsm.version.baseband<"
echo "-> Testing $TEST"
MODEM=$(getprop gsm.version.baseband)
if [ "$MODEM" != "I927UCLJ3" ];then
    echo -e "WARNING: While processing test $TEST. Found: $MODEM.\n"
else
    END_VAR "$TEST" "$MODEM"
fi

echo "\n****************************************************"
TEST=">easy install: root<"
echo "-> Testing $TEST"
ROOT=$(su -c echo "rooted")
if [ "$ROOT" != "rooted" ];then
    F_WARN "$TEST" "$ROOT"
else
    END_VAR "$TEST" "$ROOT"
fi

echo "\n****************************************************"
TEST=">BT-fix startmessage<"
echo "-> Testing $TEST"
dmesg |grep "sedi_btFCdetect.sh.*starting" | tail -n 3 > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">BT-fix finishmessage<"
echo "-> Testing $TEST"
dmesg |grep "sedi_btFCdetect.sh.*finished" | tail -n 3 > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">BT-fix indicator file<"
echo "-> Testing $TEST"
BTIPATH=$(dirname /data/misc/bluetoothd/*/.)
BTINDI="$BTIPATH/btfix.indicator"
if [ ! -f "$BTINDI" ];then
    echo $BTINDI
    F_FAIL "$TEST"
else
    END_VAR "$TEST" "$BTINDI"
fi

echo "\n****************************************************"
TEST=">/preload/.sediROM<"
echo "-> Testing $TEST"
SEDIDIR="/preload/.sediROM"
if [ ! -d "$SEDIDIR" ];then
    stat -t $SEDIDIR
    F_FAIL "$TEST"
else
    END_VAR "$TEST"
    stat -t $SEDIDIR
fi

echo "\n****************************************************"
TEST=">dir-moved-2-preload.txt<"
echo "-> Testing $TEST"
SFILE="/preload/.sediROM/dir-moved-2-preload.txt"
if [ ! -f "$SFILE" ];then
    stat -t $SFILE
    F_FAIL "$TEST"
else
    END_VAR "$TEST"
    stat -t $SFILE
fi

echo "\n****************************************************"
TEST=">initialized state<"
echo "-> Testing $TEST"
IFILE="/preload/.sediROM/.initialized"
if [ ! -f "$IFILE" ];then
    stat -t "$IFILE"
    F_FAIL "$TEST"
else
    END_VAR "$TEST"
    stat -t "$IFILE"
fi

echo "\n****************************************************"
TEST=">remountrw<"
echo "-> Testing $TEST"
remountrw /system > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">remountro<"
echo "-> Testing $TEST"
remountro /system > /sdcard/test
if [ $? -ne 0 ];then
    cat /sdcard/test
    F_FAIL "$TEST"
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">EFS backup<"
echo "-> Testing $TEST"
ls /sdcard/efs_$(date +%F)_* 2>&1 > /sdcard/test
if [ $? -ne 0 ];then
    ls /sdcard/efs_$(getprop ro.build.display.id)* 2>&1 > /sdcard/test
    if [ $? -ne 0 ];then
        cat /sdcard/test
        F_FAIL "$TEST"
    else
        F_WARN "$TEST" "matching $(ls /sdcard/efs_$(getprop ro.build.display.id)* )"
    fi
else
    END_FILE "$TEST"
fi

echo "\n****************************************************"
TEST=">init.d dir perm<"
echo "-> Testing $TEST"
EISTAT=$(stat -c %a /etc/init.d/ )
if [ "$EISTAT" -ne 750 ];then
    echo "$TEST: $EISTAT"
    F_FAIL "$TEST"
else
    END_VAR "$TEST" "$EISTAT"
fi


echo "\n****************************************************"
TEST=">init.d file perms<"
echo "-> Testing $TEST"
WORLDRW=$(find /etc/init.d -perm -o+w|wc -l)
if [ "$WORLDRW" -ne 0 ];then
    echo "$TEST: $WORLDRW"
    F_FAIL "$TEST"
else
    END_VAR "$TEST" "$(find /etc/init.d -perm -o+w)"
fi

echo "\n****************************************************"
TEST=">init.d file owner<"
echo "-> Testing $TEST"
IUSER=$(ls -la /etc/init.d |grep -v root|wc -l)
if [ "$IUSER" -ne 0 ];then
    ls -la /etc/init.d |grep -v root
    F_FAIL "$TEST"
else
    END_VAR "$TEST" "$(ls -la /etc/init.d |grep -v rootxxx)"
fi



echo -e "\n-------------------------------------------------------\n"
# Created by http://patorjk.com/software/taag/#p=display&h=1&f=Ivrit&t=sediROM%3A
# font = Ivrit
echo '       
                 _  _  ____    ___   __  __ 
  ___   ___   __| |(_)|  _ \  / _ \ |  \/  |
 / __| / _ \ / _` || || |_) || | | || |\/| |
 \__ \|  __/| (_| || ||  _ < | |_| || |  | |
 |___/ \___| \__,_||_||_| \_\ \___/ |_|  |_|
                                     
'
echo 
echo -e "Test suite summary:\n"
echo -e "\tOK count: $(grep -c OK $LOG)"
echo -e "\tWARNING count: $(grep -c WARNING $LOG)"
echo -e "\tERROR count: $(grep -c ERROR $LOG)"
echo -e "\nTest suite finished! Have a nice day ;-)\n\n"

echo -e "Test suite summary:\n" >> $LOG
echo -e "\tOK count: $(grep -c OK $LOG)" >> $LOG
echo -e "\tWARNING count: $(grep -c WARNING $LOG)" >> $LOG
echo -e "\tERROR count: $(grep -c ERROR $LOG)" >> $LOG
echo -e "\nTest suite finished $(date +"%F %T")! Have a nice day ;-)" >> $LOG

